src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -763,32 +807,32 @@ void GoogleCalendarSyncAdaptor::updateLocalCalendarNotebooks(int accountId, cons
             QString currDeviceCalendarId = notebook->pluginName().mid(7);
             if (m_serverCalendarIdToCalendarInfo[accountId].contains(currDeviceCalendarId)) {
                 // the server-side calendar exists on the device.
-                // we don't need to purge it, but we may need to update its summary/color details.
-                deviceCalendarIds.append(currDeviceCalendarId);
-                if (notebook->name() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).summary
-                        || notebook->color() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).color
-                        || notebook->description() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).description
-                        || notebook->isReadOnly()) {
-                    // calendar information changed server-side.
-                    notebook->setIsReadOnly(false);
-                    notebook->setName(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).summary);
-                    notebook->setColor(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).color);
-                    notebook->setDescription(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).description);
-                    m_storage->updateNotebook(notebook);
-                    m_storageNeedsSave = true;
+                if (needCleanSync) {
+                    // we are performing a clean sync cycle.
+                    // we will eventually delete and then insert this notebook.
+                    SOCIALD_LOG_DEBUG("queuing clean sync of local calendar" << notebook->name() << currDeviceCalendarId << "for Google account:" << accountId);
+                    m_serverCalendarIdToCalendarInfo[accountId][currDeviceCalendarId].change = GoogleCalendarSyncAdaptor::CleanSync;
```

Comment:
Well, the clean sync at least sounds relatively easy to get. I'm not still familiar with many aspects of sync stack, so can be myself overlooking, but in case sync is aborted because of network error (or similar) wouldn't it be possible to just avoid updating last sync time stamp, and avoiding marking next sync to be a clean one? Maybe even postponing the time stamp update until all changes are pushed too. Next sync would then proceed with time stamp of last fully successful one. If there were partial updates in between, these would just update to same data, i.e. make no changes(?)

On second, maybe I now understood: so if there's a case where clean sync is needed, it's better to be done only if full repopulation of the recreated notebook is possible. Makes sense. Though if clean sync could be avoided more, not so crucial anymore, I guess.